packages/client/src/actions/buyToken.ts

```diff
@@ -50,15 +50,19 @@ export async function buyToken(data: Data) {
     // Construct a URL object for the `/execute/buy` endpoint
     const params: BuyTokenPathParameters = {
       taker: taker,
-      source: client.source || "",
+      source: client.source || '',
       ...options,
     }
 
-    tokens?.forEach(
-      (token, index) =>
-        //@ts-ignore
-        (params[`tokens[${index}]`] = `${token.contract}:${token.tokenId}`)
-    )
+    if (tokens.length === 1 && options.quantity) {
+      params.token = `${tokens[0].contract}:${tokens[0].tokenId}`
+    } else {
```

Comment:
kinda find this to be a weird interface in general. feel like we should be able to just pass in an array of tokens that can include 1155 and 721 tokens.